pkg/cmd/pkg/push.go

```diff
@@ -0,0 +1,176 @@
+package pkg
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/go-buildkite/v3/buildkite"
+	"github.com/kr/pretty"
+	"github.com/oleiade/reflections"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type newPackageConfig struct {
+	RegistrySlug string `flag:"registry"`
+	FilePath     string `flag:"file"`
+	FileName     string `flag:"file-name"`
+}
+
+var isStdInReadableFunc = isStdinReadable
+
+func isStdinReadable() (bool, error) {
+	stat, err := os.Stdin.Stat()
+	if err != nil {
+		return false, fmt.Errorf("failed to stat stdin: %w", err)
+	}
+
+	readable := (stat.Mode() & os.ModeCharDevice) == 0
+	return readable, nil
+}
+
+func NewCmdPackagePush(f *factory.Factory) *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "push --registry <registry> {--file <file> | --file-name <filename> -}",
+		Short: "Push a new package to Buildkite Packages",
+		Example: heredoc.Doc(`
+			$ bk package push --registry my-registry --file my-package.tar.gz
+			$ cat my-package.tar.gz | bk package push --registry my-registry --file-name my-package.tar.gz - # Pass package via stdin, note hyphen as the argument
+		`),
+		Args:      cobra.MaximumNArgs(1),
+		ValidArgs: []string{"-"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg, err := populateFlags[newPackageConfig](cmd.PersistentFlags())
+			if err != nil {
+				return fmt.Errorf("failed to populate flags: %w", err)
+			}
+
+			if err := validateConfig(cfg, args); err != nil {
+				return fmt.Errorf("failed to validate flags and args: %w", err)
+			}
+
+			var from io.Reader
+			switch {
+			case cfg.FilePath != "":
+				file, err := os.Open(cfg.FilePath)
+				if err != nil {
+					return fmt.Errorf("couldn't open file %s: %w", cfg.FilePath, err)
+				}
+
+				defer file.Close()
+
+				from = file
+			case len(args) != 0 && args[0] == "-":
+				from = os.Stdin
+			}
+
+			r, w := io.Pipe()
+			go func() {
+				if _, err := io.Copy(w, from); err != nil {
+					w.CloseWithError(fmt.Errorf("failed to read package from input: %w", err))
+				}
+				w.Close()
+			}()
+
+			pkg, _, err := f.RestAPIClient.PackagesService.Create(f.Config.OrganizationSlug(), cfg.RegistrySlug, buildkite.CreatePackageInput{
+				Filename: cfg.FileName,
+				Package:  r,
+			})
+			if err != nil {
+				return fmt.Errorf("making create package API request: %w", err)
+			}
+
+			fmt.Println("Created package:")
+			pretty.Println(pkg)
+			return nil
+		},
+	}
+
+	cmd.PersistentFlags().StringP("organization", "o", "", "The organization to create the package in")
```

Comment:
This isnt how the CLI works. Org is pulled out of config. I'm not sure we want to have a flag for it

Reply:
oh shoot you're totally right! this flag isn't even used, it's from a previous iteration from before i realised that org is stored. i'll remove this line